packages/core/src/App/Components/Routes/language-link.tsx

```diff
@@ -49,8 +49,8 @@ const LanguageLink = observer(({ is_clickable = false, lang, toggleModal }: TLan
                     data-testid='dt_settings_language_button'
                     id={`dt_settings_${lang}_button`}
                     key={lang}
-                    onClick={() => {
-                        changeSelectedLanguage(lang);
+                    onClick={async () => {
```

Comment:
Hi @likhith-deriv although this looks fine but can we make functions like this

`const handleClick = async (lang) => {
    await changeSelectedLanguage(lang);
    switchLanguage(lang);
    if (toggleModal) {
        toggleModal();
    }
};

// JSX
onClick={() => handleClick(lang)}`

Not sure if this will work in your case or not 😬 

this might be little more readable 👀

Reply:
Hi @fasihali-deriv, it will work.
Thanks. I'll make a note and incorporate it while making changes to this component